[Nicosia] Simplify NicosiaPlatformLayer

carlosgcampos opened the pull request:
#### ee8fbcb55b4a9151078a432d1c19073ae43b8010
<pre>
[Nicosia] Simplify NicosiaPlatformLayer
<a href="https://bugs.webkit.org/show_bug.cgi?id=263775">https://bugs.webkit.org/show_bug.cgi?id=263775</a>

Reviewed by Alejandro G. Castro.

We currently have generic classes that are created with a factory that
creates the actual implementation. We only have one implementation, so
we can just remove the unnecessary abstraction to save heap allocations,
avoid casts and make the code easier to read.

There&apos;s no change in behavior.

* Source/WebCore/page/scrolling/nicosia/ScrollingTreeFixedNodeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeFixedNodeNicosia.h:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeFrameScrollingNodeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeOverflowScrollProxyNodeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeOverflowScrollProxyNodeNicosia.h:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreePositionedNodeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreePositionedNodeNicosia.h:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeStickyNodeNicosia.h:
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::MediaPlayerPrivateGStreamer):
(WebCore::MediaPlayerPrivateGStreamer::~MediaPlayerPrivateGStreamer):
(WebCore::MediaPlayerPrivateGStreamer::pushTextureToCompositor):
(WebCore::MediaPlayerPrivateGStreamer::pushDMABufToCompositor):
(WebCore::MediaPlayerPrivateGStreamer::triggerRepaint):
(WebCore::MediaPlayerPrivateGStreamer::flushCurrentBuffer):
(WebCore::MediaPlayerPrivateGStreamer::pushNextHolePunchBuffer):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaBackingStoreTextureMapperImpl.cpp.
(Nicosia::BackingStore::tiledBackingStoreHasPendingTileCreation):
(Nicosia::BackingStore::createTile):
(Nicosia::BackingStore::updateTile):
(Nicosia::BackingStore::removeTile):
(Nicosia::BackingStore::flushUpdate):
(Nicosia::BackingStore::takeUpdate):
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaBackingStoreTextureMapperImpl.h.
* Source/WebCore/platform/graphics/nicosia/NicosiaCompositionLayer.h: Copied from Source/WebCore/platform/graphics/nicosia/NicosiaPlatformLayer.h.
* Source/WebCore/platform/graphics/nicosia/NicosiaContentLayer.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaContentLayerTextureMapperImpl.cpp.
(Nicosia::ContentLayer::create):
(Nicosia::ContentLayer::ContentLayer):
(Nicosia::ContentLayer::~ContentLayer):
(Nicosia::ContentLayer::invalidateClient):
(Nicosia::ContentLayer::flushUpdate):
(Nicosia::ContentLayer::swapBuffersIfNeeded):
* Source/WebCore/platform/graphics/nicosia/NicosiaContentLayer.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaContentLayerTextureMapperImpl.h.
* Source/WebCore/platform/graphics/nicosia/NicosiaGCGLANGLELayer.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLANGLELayer.cpp.
(Nicosia::GCGLANGLELayer::swapBuffersIfNeeded):
(Nicosia::GCGLANGLELayer::GCGLANGLELayer):
(Nicosia::GCGLANGLELayer::~GCGLANGLELayer):
* Source/WebCore/platform/graphics/nicosia/NicosiaGCGLANGLELayer.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLANGLELayer.h.
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBacking.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaImageBackingTextureMapperImpl.cpp.
(Nicosia::ImageBacking::flushUpdate):
(Nicosia::ImageBacking::takeUpdate):
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBacking.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaImageBackingTextureMapperImpl.h.
(Nicosia::ImageBacking::create):
(Nicosia::ImageBacking::layerState):
(Nicosia::ImageBacking::compositionState):
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBackingStore.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaImageBackingStore.cpp.
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBackingStore.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaImageBackingStore.h.
(Nicosia::ImageBackingStore::backingStoreState):
(Nicosia::ImageBackingStore::CompositionState::CompositionState):
(Nicosia::ImageBackingStore::compositionState):
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBufferPipe.cpp:
(Nicosia::NicosiaImageBufferPipeSource::NicosiaImageBufferPipeSource):
(Nicosia::NicosiaImageBufferPipeSource::~NicosiaImageBufferPipeSource):
(Nicosia::NicosiaImageBufferPipeSource::handle):
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBufferPipe.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaPlatformLayer.cpp: Removed.
* Source/WebCore/platform/graphics/nicosia/NicosiaPlatformLayer.h:
(Nicosia::PlatformLayer::PlatformLayer):
(Nicosia::CompositionLayer::Impl::isTextureMapperImpl const): Deleted.
(Nicosia::CompositionLayer::create): Deleted.
(Nicosia::CompositionLayer::impl const): Deleted.
(Nicosia::CompositionLayer::LayerState::Flags::Flags): Deleted.
(Nicosia::CompositionLayer::flushState): Deleted.
(Nicosia::CompositionLayer::commitState): Deleted.
(Nicosia::CompositionLayer::accessPending): Deleted.
(Nicosia::CompositionLayer::accessCommitted): Deleted.
(Nicosia::ContentLayer::Impl::isTextureMapperImpl const): Deleted.
(Nicosia::ContentLayer::create): Deleted.
(Nicosia::ContentLayer::impl const): Deleted.
(Nicosia::BackingStore::Impl::isTextureMapperImpl const): Deleted.
(Nicosia::BackingStore::create): Deleted.
(Nicosia::BackingStore::impl const): Deleted.
(Nicosia::ImageBacking::Impl::isTextureMapperImpl const): Deleted.
(Nicosia::ImageBacking::create): Deleted.
(Nicosia::ImageBacking::impl const): Deleted.
* Source/WebCore/platform/graphics/nicosia/NicosiaScene.cpp:
* Source/WebCore/platform/graphics/nicosia/NicosiaSceneIntegration.cpp:
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaCompositionLayerTextureMapperImpl.cpp: Removed.
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaCompositionLayerTextureMapperImpl.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::CoordinatedGraphicsLayer):
(WebCore::CoordinatedGraphicsLayer::updatePlatformLayer):
(WebCore::CoordinatedGraphicsLayer::flushCompositingStateForThisLayerOnly):
(WebCore::CoordinatedGraphicsLayer::updateContentBuffers):
(WebCore::CoordinatedGraphicsLayer::purgeBackingStores):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h:
* Source/WebKit/Shared/CoordinatedGraphics/CoordinatedGraphicsScene.cpp:
(WebKit::texmapLayer):
(WebKit::updateBackingStore):
(WebKit::updateImageBacking):
(WebKit::removeLayer):
(WebKit::CoordinatedGraphicsScene::updateSceneState):
(WebKit::compositionLayerImpl): Deleted.
(WebKit::contentLayerImpl): Deleted.
(WebKit::backingStoreImpl): Deleted.
(WebKit::imageBackingImpl): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.cpp:
(WebKit::CompositingCoordinator::flushPendingLayerChanges):

Canonical link: <a href="https://commits.webkit.org/269998@main">https://commits.webkit.org/269998@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/14e74c308d4d048175aafe66c11b24fef1ce1923

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23815 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1927 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24911 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/25964 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21962 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24089 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3529 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24314 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22485 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24058 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1474 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20587 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26556 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1244 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21498 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27752 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21719 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21778 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25535 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1197 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18887 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1199 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1608 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3093 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1515 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
LGTM! Great simplification of the code.

carlosgcampos committed:
[Nicosia] Simplify NicosiaPlatformLayer
https://bugs.webkit.org/show_bug.cgi?id=263775

Reviewed by Alejandro G. Castro.

We currently have generic classes that are created with a factory that
creates the actual implementation. We only have one implementation, so
we can just remove the unnecessary abstraction to save heap allocations,
avoid casts and make the code easier to read.

There's no change in behavior.

* Source/WebCore/page/scrolling/nicosia/ScrollingTreeFixedNodeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeFixedNodeNicosia.h:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeFrameScrollingNodeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeOverflowScrollProxyNodeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeOverflowScrollProxyNodeNicosia.h:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreePositionedNodeNicosia.cpp:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreePositionedNodeNicosia.h:
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeStickyNodeNicosia.h:
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::MediaPlayerPrivateGStreamer):
(WebCore::MediaPlayerPrivateGStreamer::~MediaPlayerPrivateGStreamer):
(WebCore::MediaPlayerPrivateGStreamer::pushTextureToCompositor):
(WebCore::MediaPlayerPrivateGStreamer::pushDMABufToCompositor):
(WebCore::MediaPlayerPrivateGStreamer::triggerRepaint):
(WebCore::MediaPlayerPrivateGStreamer::flushCurrentBuffer):
(WebCore::MediaPlayerPrivateGStreamer::pushNextHolePunchBuffer):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaBackingStoreTextureMapperImpl.cpp.
(Nicosia::BackingStore::tiledBackingStoreHasPendingTileCreation):
(Nicosia::BackingStore::createTile):
(Nicosia::BackingStore::updateTile):
(Nicosia::BackingStore::removeTile):
(Nicosia::BackingStore::flushUpdate):
(Nicosia::BackingStore::takeUpdate):
* Source/WebCore/platform/graphics/nicosia/NicosiaBackingStore.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaBackingStoreTextureMapperImpl.h.
* Source/WebCore/platform/graphics/nicosia/NicosiaCompositionLayer.h: Copied from Source/WebCore/platform/graphics/nicosia/NicosiaPlatformLayer.h.
* Source/WebCore/platform/graphics/nicosia/NicosiaContentLayer.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaContentLayerTextureMapperImpl.cpp.
(Nicosia::ContentLayer::create):
(Nicosia::ContentLayer::ContentLayer):
(Nicosia::ContentLayer::~ContentLayer):
(Nicosia::ContentLayer::invalidateClient):
(Nicosia::ContentLayer::flushUpdate):
(Nicosia::ContentLayer::swapBuffersIfNeeded):
* Source/WebCore/platform/graphics/nicosia/NicosiaContentLayer.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaContentLayerTextureMapperImpl.h.
* Source/WebCore/platform/graphics/nicosia/NicosiaGCGLANGLELayer.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLANGLELayer.cpp.
(Nicosia::GCGLANGLELayer::swapBuffersIfNeeded):
(Nicosia::GCGLANGLELayer::GCGLANGLELayer):
(Nicosia::GCGLANGLELayer::~GCGLANGLELayer):
* Source/WebCore/platform/graphics/nicosia/NicosiaGCGLANGLELayer.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaGCGLANGLELayer.h.
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBacking.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaImageBackingTextureMapperImpl.cpp.
(Nicosia::ImageBacking::flushUpdate):
(Nicosia::ImageBacking::takeUpdate):
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBacking.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaImageBackingTextureMapperImpl.h.
(Nicosia::ImageBacking::create):
(Nicosia::ImageBacking::layerState):
(Nicosia::ImageBacking::compositionState):
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBackingStore.cpp: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaImageBackingStore.cpp.
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBackingStore.h: Renamed from Source/WebCore/platform/graphics/nicosia/texmap/NicosiaImageBackingStore.h.
(Nicosia::ImageBackingStore::backingStoreState):
(Nicosia::ImageBackingStore::CompositionState::CompositionState):
(Nicosia::ImageBackingStore::compositionState):
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBufferPipe.cpp:
(Nicosia::NicosiaImageBufferPipeSource::NicosiaImageBufferPipeSource):
(Nicosia::NicosiaImageBufferPipeSource::~NicosiaImageBufferPipeSource):
(Nicosia::NicosiaImageBufferPipeSource::handle):
* Source/WebCore/platform/graphics/nicosia/NicosiaImageBufferPipe.h:
* Source/WebCore/platform/graphics/nicosia/NicosiaPlatformLayer.cpp: Removed.
* Source/WebCore/platform/graphics/nicosia/NicosiaPlatformLayer.h:
(Nicosia::PlatformLayer::PlatformLayer):
(Nicosia::CompositionLayer::Impl::isTextureMapperImpl const): Deleted.
(Nicosia::CompositionLayer::create): Deleted.
(Nicosia::CompositionLayer::impl const): Deleted.
(Nicosia::CompositionLayer::LayerState::Flags::Flags): Deleted.
(Nicosia::CompositionLayer::flushState): Deleted.
(Nicosia::CompositionLayer::commitState): Deleted.
(Nicosia::CompositionLayer::accessPending): Deleted.
(Nicosia::CompositionLayer::accessCommitted): Deleted.
(Nicosia::ContentLayer::Impl::isTextureMapperImpl const): Deleted.
(Nicosia::ContentLayer::create): Deleted.
(Nicosia::ContentLayer::impl const): Deleted.
(Nicosia::BackingStore::Impl::isTextureMapperImpl const): Deleted.
(Nicosia::BackingStore::create): Deleted.
(Nicosia::BackingStore::impl const): Deleted.
(Nicosia::ImageBacking::Impl::isTextureMapperImpl const): Deleted.
(Nicosia::ImageBacking::create): Deleted.
(Nicosia::ImageBacking::impl const): Deleted.
* Source/WebCore/platform/graphics/nicosia/NicosiaScene.cpp:
* Source/WebCore/platform/graphics/nicosia/NicosiaSceneIntegration.cpp:
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaCompositionLayerTextureMapperImpl.cpp: Removed.
* Source/WebCore/platform/graphics/nicosia/texmap/NicosiaCompositionLayerTextureMapperImpl.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::CoordinatedGraphicsLayer):
(WebCore::CoordinatedGraphicsLayer::updatePlatformLayer):
(WebCore::CoordinatedGraphicsLayer::flushCompositingStateForThisLayerOnly):
(WebCore::CoordinatedGraphicsLayer::updateContentBuffers):
(WebCore::CoordinatedGraphicsLayer::purgeBackingStores):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h:
* Source/WebKit/Shared/CoordinatedGraphics/CoordinatedGraphicsScene.cpp:
(WebKit::texmapLayer):
(WebKit::updateBackingStore):
(WebKit::updateImageBacking):
(WebKit::removeLayer):
(WebKit::CoordinatedGraphicsScene::updateSceneState):
(WebKit::compositionLayerImpl): Deleted.
(WebKit::contentLayerImpl): Deleted.
(WebKit::backingStoreImpl): Deleted.
(WebKit::imageBackingImpl): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CompositingCoordinator.cpp:
(WebKit::CompositingCoordinator::flushPendingLayerChanges):

Canonical link: https://commits.webkit.org/269998@main
